esx_service/vmdk_ops.py

```diff
@@ -787,6 +790,7 @@ def get_datastore_name(datastore_url):
                       datastore_name)
```

Comment:
Not always. datastore_name can be None, or a stale one. Therefore, /vmfs/volumes/datastore_name not always exist.  If it does not exist, we refresh the datastore cache to get the updated "datastore_name".